server/graphql/general/src/queries/names.rs

```diff
@@ -91,6 +91,39 @@ pub fn get_names(
     let service_provider = ctx.service_provider();
     let service_context = service_provider.context(store_id.clone(), user.user_id)?;
 
+    // exclude the current store from the list of names
+    // if requesting for customers or suppliers
+    let filter = {
```

Comment:
Wouldn't it be better to fix data ? 

I am a bit wary that extra logic added to deal with edge cases in data will have undesirable effect on code readability.
(I don't think it's possible to have this type of setup without foot runner code ?)

Reply:
totally agree about fixing the problem at source! 
but... we have had this raised by three different people now, using three different datafiles - so while this is bad data, it is likely to come through to oms from mSupply and we do need to address somehow. This approach seems the simplest; alternative is to run data migration in mSupply to correct and raise an issue to determine how it is happening. We would still have the data issue in mSupply installs which are older than the patched version though.

Reply:
>  three different people now, using three different datafiles.

Hmm, in that case yeah, we should do something about this. I still would suggest fixing the data, but we can do it on omSupply end:
1 - During sync translation, skip name_store_joins where name_id belongs to store_id.
2 - Data migration to remove those (single sql statement in migrations)

Above might be slightly more code but I think it's more direct way to solve the issue. 

If we want to go down filter out path, I think we should do this at database layer and exclude name_store_join with name_id that belong to store_id: https://github.com/openmsupply/open-msupply/compare/986-example

This result in sql statement like this:

```
SELECT * FROM name
LEFT OUTER JOIN name_store_join 
    ON name_store_join.name_id = name.id
    AND name_store_join.store_id = 'abc'
    AND name_store_join.name_id NOT IN (SELECT store.name_id from store where store.id = 'abc')
```

And example of how this would work:

https://www.db-fiddle.com/f/7axvQz7iYhUGuyxwwmTj23/1

Lastly, another reason why I like checking data when it comes in rather then filtering out when it's used is:
duplicate name_store_joins may exists for the same store and name, this would cause duplicates showing up in the query

Reply:
cool. How's this? d562db2c3cf715f68b73c83abe81378adf2db667

Reply:
That looks great @mark-prins, thanks!

There is still one little thing:

Do we need to check for is_supplier and is_customer ? There is another translator that check if name.is_customer or name.is_supplier was changed (`name_to_name_store_join`) and updates name_store_join. I think this would mean if self referencing name_store_join exists but the name is not customer or supplier, it won't be deleted, but when name is made a customer or supplier it would set that on name_store_join.. 

I think it's safe to delete any self referencing name_store_joins (even if they are not customer or suppier)

Reply:
Was just being (probably overly) cautious. Happy to remove 'em all. b1099418a8f9a006a3938e20412f9f9dfd507df5